app/models/notification_type.rb

```diff
@@ -0,0 +1,38 @@
+class NotificationType < ApplicationRecord
+  AUDIENCE_USER = 'user'.freeze
+  AUDIENCE_TENANT = 'tenant'.freeze
+  AUDIENCE_GLOBAL = 'global'.freeze
+  has_many :notifications
+  validates :message, :presence => true
+  validates :level, :inclusion => { :in => %w(success error warning info) }
+  validates :audience, :inclusion => { :in => [AUDIENCE_USER, AUDIENCE_TENANT, AUDIENCE_GLOBAL] }
+
+  def subscriber_ids(subject, initiator)
```

Comment:
Not sure about this, a `subscribers` [collection proxy](http://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html) can be sometimes more versatile. Then you can call the `#pluck(:id)` on it externally.